week-1-ts-ecosystem/w01e04-light-me-up/src/main.ts

```diff
@@ -8,14 +12,20 @@
  * Dodatkowo — rzucać błąd, jeśli zapas mocy się wyczerpie.
  * */
 
-class PowerSource {
-  private energySupply = 100
+// ZADANIE 1
 
-  consume(energy) {
-    this.energySupply -= energy
-  }
-}
+const powerSource = new PowerSource();
+const errorHandler = new ErrorHandler();
+const lightBulb1 = new LightBulb(powerSource, errorHandler);
+const lightBulb2 = new LightBulb(powerSource, errorHandler);
+const lightBulb3 = new LightBulb(powerSource, errorHandler);
+const lightBulb4 = new LightBulb(powerSource, errorHandler);
+const lightBulb5 = new LightBulb(powerSource, errorHandler);
+const lightBulb6 = new LightBulb(powerSource, errorHandler);
 
-class LightBulb {
-  protected readonly powerConsumption = 20
-}
+lightBulb1.light();
+lightBulb2.light();
+lightBulb3.light();
+lightBulb4.light();
+lightBulb5.light();
+lightBulb6.light(); //poleci exception
```

Comment:
A jak byś rozwiązał ten problem @kdrzymalski  - żeby to tutaj łapać Exceptiony ???

https://community.ahoy.so/c/zadania/ukryte-znaczenie-zadania-w01e04-zarowka

Reply:
Rozumiem że try catch nie zadziała?
Co powiesz na wykorzystanie window.onerror
Chyba że idę w zupełnie złym kierunku?

Reply:
Co do łapania takich errorów w Node.js (express) zawsze pisałem try catch wewnątrz callbacku jak tutaj a globalne errory łapałem w specjalnym middleware z 4 argumentami, gdzie pierwszy argument to jest err i tak Express łapał mi errory.
Czy takie podejście jest Twoim zdaniem wystarczające?

Reply:
Tak, jeśli zrobisz na `callback`'ach te errory to jedno z dwóch rozwiązań (drugie to `Promise`) - które są ok.

`window.onerror` w **Node.js** to będzie:

```javascript
process.on('uncaughtException', (error) => {
  console.log(errror);
});
```

i to też powinno występować w tym kodzie.
Po to żeby program "nie wykolejał się" jeśli wystąpi nie złapany wyjątek.
Ten kod powyżej z `uncaughtException` powinien być standardem jak piszesz serwer czy inny back-end w Node.js

[uncaughtException](https://nodejs.org/api/process.html#event-uncaughtexception)